### PR TITLE
TF: XLA repetition penalty

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

This PR adds our first XLA-compatible TF logit processor, as well as corresponding tests. Since this is the first of a series of small (but similar) PRs, I'd like to request a more thorough review, so the remaining ones are quick.

More specifically, this PR makes three changes:
1. Rewrites the TF repetition penalty processor so as to be XLA-compatible;
2. Adds XLA tests for the processor;
3. Since the test mentioned in 2. was a near copy/paste of the non-XLA test, I've decided to split the test into three parts to improve code reuse and reduce errors from ad hoc edits (as the first and last part can be reused in the two versions of the test, XLA and non-XLA)
    - get inputs
    - run the processor
    - check the output

### CODE REVIEW

File: tests/generation/test_generation_tf_logits_process.py [hunk 1]
```diff
@@ -114,17 +114,32 @@ def test_repetition_penalty_dist_process(self):
...
         self.assertAlmostEqual(scores[1, 5].numpy(), (4 / vocab_size) / 2)
+        self.assertAlmostEqual(scores[0, 2].numpy(), (1 / vocab_size))  # unused tokens should see no change
+
+    def test_repetition_penalty_dist_process(self):
+        vocab_size, cur_len, input_ids, scores = self._get_repetition_penalty_inputs()
+        rep_penalty_proc = TFRepetitionPenaltyLogitsProcessor(penalty=2.0)
+        scores = rep_penalty_proc(input_ids, tf.identity(scores), cur_len)
+        self._check_repetition_penalty_outputs(scores, vocab_size)
+
+    @unittest.skipIf(not get_gpu_count(), "XLA not reliable on CPU")
```
Comment (patrickvonplaten) [inline tests/generation/test_generation_tf_logits_process.py:136]: This should really be solved IMO. Ok to have it like this for now, but the problem is that it's never run on Circle CI so we would only find it when merged to master or for the daily tests.

Comment (patrickvonplaten) [reply tests/generation/test_generation_tf_logits_process.py:136]: Could we maybe add a big `# TODO` here to clean it up

Comment (gante) [reply tests/generation/test_generation_tf_logits_process.py:136]: Agreed, it is highly undesirable. Adding the TODO on all related `skipIf`.

Comment (gante) [reply tests/generation/test_generation_tf_logits_process.py:136]: actually -- removing the skips here, as the issue seems to be exclusively around the softmax (but adding the TODO to the models)

**Review by patrickvonplaten — APPROVED** [review]

**Review by Rocketknight1 — APPROVED** [review]
This looks like a good port of the original numpy code!

(But I still think logit penalties should be additive rather than multiplicative)

**Comment by Rocketknight1** [comment]
Thinking about it more, a multiplicative logit penalty really doesn't work, right? Even if we use the reciprocal when the logit is negative, the scale of the penalty depends on the logit's distance from 0. For example, a logit in the range -0.1 to +0.1 will barely be moved by the penalty term, but such logits usually have quite a high probability of being chosen, because most logits are large and negative.

**Comment by gante** [comment]
(merging as the main goal was to port to XLA but, by all means, continue the discussion :) )
